plugins/woocommerce-blocks/assets/js/editor-components/default-notice/index.tsx

```diff
@@ -42,7 +42,7 @@ export function DefaultNotice( { block }: { block: string } ) {
 			postPublished: isCurrentPostPublished(),
 			currentPostId: getCurrentPostId(),
 		};
-	}, [] );
+	} );
```

Comment:
I'm not sure this is the correct fix. Currently, with `[]`, the code runs only when the component mounts. If we remove `[]`, the selector will be triggered on every render.

We have a couple of options:
- Keep `[]` and add an ESLint comment to disable the warning.
- Replace `[]` with `[ORIGINAL_PAGE_ID, block]`.

`ORIGINAL_PAGE_ID` is a constant, so it's safe to include, and `block` is a prop, so the function will re-invoke whenever the `block` prop changes.

We should check with the ex-Mothra team to understand if there was a specific reason they originally added `[]`.

Reply:
I reverted to change to an empty array but I won't disable the warning because I'm not sure about the expected outcome. Once merged, I'll create an issue for Rubik to assess the remaining three issues so they can either make adjustments to hooks dependencies or disable warning if the current state is expected.

Thanks a lot for catching this @gigitux!

Reply:
> I reverted the change to an empty array but I won't disable the warning because I'm not sure about the expected outcome. 

Makes sense! 💯